src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -160,7 +161,12 @@ public override void SetLength(long value)
         /// </summary>
         public override void Flush()
         {
-            _stream.Flush();
+            // Can sometimes get Pipe broken errors from flushing a PipeStream.
+            // PipeStream.Flush() also doesn't do anything, anyway.
+            if (!(_stream is PipeStream))
```

Comment:
Currently Sqlclient doesn't use any other wrapped around PipeStream except NamedPipeClientStream. In terms of performance, wouldn't  it be better to check for the type and avoid the flush or wait for an exception and omit it?